### PR TITLE
feat: display used config path when logging level=log

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**

feat

**Did you add tests for your changes?**
yes

**If relevant, did you update the documentation?**
NA

**Summary**
- Displays the used config when infra logging level is log, helpful to know which config is in use when there are multiple

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

File: packages/webpack-cli/lib/plugins/CLIPlugin.js [hunk 1]
```diff
@@ -41,6 +41,13 @@ class CLIPlugin {
         const pluginName = 'webpack-cli';
         const getCompilationName = () => (compiler.name ? ` '${compiler.name}'` : '');
 
+        const { configPath } = this.options;
+        if (configPath) {
+            compiler.hooks.run.tap(configPath, () => {
+                this.logger.log(`Using config ${configPath}`);
+            });
+        }
```
Comment (alexander-akait) [inline packages/webpack-cli/lib/plugins/CLIPlugin.js:49]: Let's use this logic in hook below

**Comment by alexander-akait** [comment]
@anshumanv can you rebase and fix the problem

**Comment by alexander-akait** [comment]
We should run message on `watch` and `serve` too

**Comment by anshumanv** [comment]
yep lets do it, more issues with test will fix them today

**Comment by alexander-akait** [comment]
Please rebase fork

File: test/core-flags/infrastructure-logging.test.js [hunk 2]
```diff
@@ -27,4 +28,56 @@ describe('infrastructure logging related flag', () => {
...
+            '--config',
+            'webpack.cache.config.js',
+            '--infrastructure-logging-level',
+            'log',
+        ]);
+
+        const configPath = resolve(__dirname, './webpack.cache.config.js');
+        expect(stderr).toContain(`Using config ${configPath}`);
+        expect(stdout).toContain(`level: 'log'`);
+    });
```
Comment (alexander-akait) [inline test/core-flags/infrastructure-logging.test.js:82]: Wrong place for tests here, in `core-flags` we tests `flags`, put each test to `test/build/basic`/`test/watch/basic`/`test/serve/basic` directories

Comment (anshumanv) [reply test/core-flags/infrastructure-logging.test.js:82]: moving

File: test/build/basic/basic.test.js [hunk 3]
```diff
@@ -140,4 +140,14 @@ describe('bundle command', () => {
         expect(stderr).toContain("Run 'webpack --help' to see available commands and options");
         expect(stdout).toBeFalsy();
     });
+
+    it('should log supplied config when logging level is log', () => {
+        if (isWebpack5) {
```
Comment (alexander-akait) [inline test/build/basic/basic.test.js:145]: Should worn on v4 too

Comment (anshumanv) [reply test/build/basic/basic.test.js:145]: worn?

Comment (alexander-akait) [reply test/build/basic/basic.test.js:145]: work :smile:

Comment (anshumanv) [reply test/build/basic/basic.test.js:145]: This is introduced in core flag, was not supported before as flag

Comment (alexander-akait) [reply test/build/basic/basic.test.js:145]: It is just logger, please remove `isWebpack5`

Comment (anshumanv) [reply test/build/basic/basic.test.js:145]: won't work 
image

File: test/core-flags/infrastructure-logging.test.js [hunk 4]
```diff
@@ -27,4 +28,15 @@ describe('infrastructure logging related flag', () => {
...
+    it('should log used default config when level is log', () => {
+        const { exitCode, stderr, stdout } = run(__dirname, ['--infrastructure-logging-level', 'log']);
+
+        expect(exitCode).toBe(0);
+        const configPath = resolve(__dirname, './webpack.config.js');
+        expect(stderr).toContain("Compilation 'compiler' starting...");
+        expect(stderr).toContain(`Using config ${configPath}`);
+        expect(stderr).toContain("Compilation 'compiler' finished");
+        expect(stdout).toContain(`level: 'log'`);
+    });
```
Comment (alexander-akait) [inline test/core-flags/infrastructure-logging.test.js:41]: We don't need this test here, we already have test for `--infrastructure-logging-level`

Comment (anshumanv) [reply test/core-flags/infrastructure-logging.test.js]: done remove

**Comment by anshumanv** [comment]
Closing - https://github.com/webpack/webpack-cli/pull/2431

**Pull request closed without merging**